govcd/vapp_test.go

```diff
@@ -137,6 +136,19 @@ func (vcd *TestVCD) Test_AddMetadata(check *C) {
 
 }
 
+func (vcd *TestVCD) Test_DeleteMetadataOnVapp(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+	// Remove metadata
+	task, err := vcd.vapp.DeleteMetadata("key")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+
```

Comment:
I recommend not only test that Task completed successful but and query vapp metadata and verify is not there anymore.

Reply:
Fixed.